### PR TITLE
Release candidate v1.0.1

### PR DESCRIPTION
**Pull request by BarryPiccinni**
This commit fixes a bug in the summary transformer. Previously,
section headings were rendered whether there were relevant questions
or not.

Now, sections which have no questions do not get transformed at all.

This has been manually tested and automated tests have been updated to include this.

### CODE REVIEW

File: lib/transformers/summary.js [hunk 1]
```diff
@@ -42,11 +42,13 @@ module.exports = ({schemaKey, schema, options, data, fullUiSchema} = {}) => {
             }
         });
 
-        summaryLists += `<h2 class="govuk-heading-l">${summaryList.title}</h2>
+        if (rows.length > 0) {
+            summaryLists += `<h2 class="govuk-heading-l">${summaryList.title}</h2>
               {{ govukSummaryList({
                 classes: 'govuk-!-margin-bottom-9',
                 rows: ${JSON.stringify(rows, null, 4)}
```
Comment (armoj) [inline lib/transformers/summary.js:49]: just a thought: Change this to:
`rows: JSON.stringify(rows)`
Is this uneeded formatting on the json?
i did a quick perf test and it is 20% slower than just doing my suggestion above.

Comment (sinclairs) [reply lib/transformers/summary.js:49]: I'm not sure if it applies here but in some situations the whitespace is required: https://github.com/CriminalInjuriesCompensationAuthority/q-transformer/blob/b0732d18c9f6bc3770c0c6e3ec33304d704b7587/lib/helpers/nunjucks/index.js#L55

Comment (BarryPiccinni) [reply lib/transformers/summary.js:49]: Yeah, the spacing is required for nunjucks to render this correctly.

File: test/q-transformer.test.js [hunk 2]
```diff
@@ -1605,7 +1605,7 @@ describe('qTransformer', () => {
...
+                        '<h2 class="govuk-heading-l">Your details</h2>\n{{ govukSummaryList({\nclasses: \'govuk-!-margin-bottom-9\',\nrows: [\n{\n"key": {\n"text": "Name",\n"classes": "govuk-!-width-one-half"\n},\n"value": {\n"html": "Mr Barry Piccinni"\n},\n"actions": {\n"items": [\n{\n"href": "/apply/applicant-enter-your-name?next=check-your-answers",\n"text": "Change",\n"visuallyHiddenText": "Name"\n}\n]\n}\n}\n]\n}) }}\n<h2 class="govuk-heading-l">Agree and submit your application</h2>\n<p class="govuk-body">By submitting this application you agree that we can share the details in it with the police. This helps us get the police information that we need to make a decision.</p>\n<p class="govuk-body">To find out more about how we handle your data <a href="https://www.gov.uk/guidance/cica-privacy-notice" target="">read our privacy notice</a>.</p>\n',
```
Comment (armoj) [inline test/q-transformer.test.js:1608]: remove personal details from here

File: test/q-transformer.test.js [hunk 3]
```diff
@@ -1634,7 +1634,36 @@ describe('qTransformer', () => {
...
...-bottom-9\',\nrows: [\n{\n"key": {\n"text": "Name",\n"classes": "govuk-!-width-one-half"\n},\n"value": {\n"html": "Mr Barry Piccinni"\n},\n"actions": {\n"items": [\n{\n"href": "/apply/applicant-enter-your-name?next=check-your-answers",\n"text": "Change",\n"visuallyHiddenText": "Name"\n}\n]\n}\n}\n]\n}) }}<h2 class="govuk-heading-l">About the crime</h2>\n{{ govukSummaryList({\nclasses: \'govuk-!-margin-bottom-9\',\nrows: [\n{\n"key": {\n"text": "When did the crime happen?",\n"classes": "govuk-!-width-one-half"\n},\n"value": {\n"html": "01 January 2019"\n},\n"actions": {\n"items": [\n{\n"href": "/apply/applicant-when-did-the-crime-happen?next=check-your-answers",\n"text": "Change",\n"visuallyHiddenText": "When did the crime happen?"\n}\n]\n}\n}\n]\n}) }}\n<h2 class="govuk-heading-l">Agree and submit your application</h2>\n<p class="govuk-body">By submitting this application you agree that we can share the details in it with the police. This helps us get the police information that we need to make a decision.</p>\n<p class="govuk-body">To find out more about how we handle your data <a href="https://www.gov.uk/guidance/cica-privacy-notice" target="">read our privacy notice</a>.</p>\n',
```
Comment (armoj) [inline test/q-transformer.test.js:1637]: Remove your name from here

File: test/q-transformer.test.js [hunk 4]
```diff
@@ -1634,7 +1634,36 @@ describe('qTransformer', () => {
...
...-!-margin-bottom-9\',\nrows: [\n{\n"key": {\n"text": "Name",\n"classes": "govuk-!-width-one-half"\n},\n"value": {\n"html": "Mr Barry Piccinni"\n},\n"actions": {\n"items": [\n{\n"href": "/apply/applicant-enter-your-name?next=check-your-answers",\n"text": "Change",\n"visuallyHiddenText": "Name"\n}\n]\n}\n}\n]\n}) }}<h2 class="govuk-heading-l">Other compensation</h2>\n{{ govukSummaryList({\nclasses: \'govuk-!-margin-bottom-9\',\nrows: [\n{\n"key": {\n"text": "Have you applied before?",\n"classes": "govuk-!-width-one-half"\n},\n"value": {\n"html": "Yes"\n},\n"actions": {\n"items": [\n{\n"href": "/apply/applicant-have-you-applied-to-us-before?next=check-your-answers",\n"text": "Change",\n"visuallyHiddenText": "Have you applied before?"\n}\n]\n}\n}\n]\n}) }}\n<h2 class="govuk-heading-l">Agree and submit your application</h2>\n<p class="govuk-body">By submitting this application you agree that we can share the details in it with the police. This helps us get the police information that we need to make a decision.</p>\n<p class="govuk-body">To find out more about how we handle your data <a href="https://www.gov.uk/guidance/cica-privacy-notice" target="">read our privacy notice</a>.</p>\n',
```
Comment (armoj) [inline test/q-transformer.test.js:1666]: remove you name from here

File: test/q-transformer.test.js [hunk 5]
```diff
@@ -1663,7 +1692,7 @@ describe('qTransformer', () => {
...
...-bottom-9\',\nrows: [\n{\n"key": {\n"text": "Name",\n"classes": "govuk-!-width-one-half"\n},\n"value": {\n"html": "Mr Barry Piccinni"\n},\n"actions": {\n"items": [\n{\n"href": "/apply/applicant-enter-your-name?next=check-your-answers",\n"text": "Change",\n"visuallyHiddenText": "Name"\n}\n]\n}\n}\n]\n}) }}<h2 class="govuk-heading-l">About the crime</h2>\n{{ govukSummaryList({\nclasses: \'govuk-!-margin-bottom-9\',\nrows: [\n{\n"key": {\n"text": "When did the crime happen?",\n"classes": "govuk-!-width-one-half"\n},\n"value": {\n"html": "01 January 2019"\n},\n"actions": {\n"items": [\n{\n"href": "/apply/applicant-when-did-the-crime-happen?next=check-your-answers",\n"text": "Change",\n"visuallyHiddenText": "When did the crime happen?"\n}\n]\n}\n}\n]\n}) }}\n<h2 class="govuk-heading-l">Agree and submit your application</h2>\n<p class="govuk-body">By submitting this application you agree that we can share the details in it with the police. This helps us get the police information that we need to make a decision.</p>\n<p class="govuk-body">To find out more about how we handle your data <a href="https://www.gov.uk/guidance/cica-privacy-notice" target="">read our privacy notice</a>.</p>\n',
```
Comment (armoj) [inline test/q-transformer.test.js:1695]: remove your name from here

**Pull request merged**